server/trace/mock_trace.go

```diff
@@ -32,24 +32,25 @@ import (
 type MockTraceServer struct {
 	cloudtrace.UnimplementedTraceServiceServer
 	mocktrace.UnimplementedMockTraceServiceServer
-	uploadedSpans     map[string]*cloudtrace.Span
+	uploadedSpanNames map[string]bool
```

Comment:
Replacing the map of `uploadedSpans` with an array and a map of just their names so we can access elements at a given index, while also checking for duplicate spans in constant time

Reply:
It's more idiomatic to use `map[string]struct{}`. Otherwise it's not clear whether there's a difference between setting a value to false vs true. It's also marginally faster than bool.

Reply:
Got it, done